### PR TITLE
saved searches: remove notifications option if not enabled

### LINKED ISSUE #22154: Remove notifications functionality from saved searches
### Context

We decided in planning code monitoring that notifications functionality for saved searches would be removed once we were confident in code monitoring's performance. I believe we've validated code monitoring as a replacement for these notifications, and now we need to remove the notifications functionality from saved searches.

_Note that saved searches themselves will continue to exist in Sourcegraph in the form of "easy re-usable search queries," but without notifications._

### Proposal 

Existing saved searches with notifications should continue to work for now. We should:

1. Remove the ability to add notifications to saved searches that aren't already using them, and update content to reflect functionality:

CleanShot 2021-06-17 at 12 11 53@2x

Note that the button label has changed and there's now a "Cancel" button as well.

2. Disable interactivity with saved search notifications that already have them configured, and prompt users to create a code monitor instead.

There's a couple of approaches to this: the slightly-more-scope approach, and the minimal scope approach.

In the minimal scope approach, we show the "Email notifications" checkbox, and show an info message telling the user to go to code monitoring. When the user unchecks the checkbox and saves the view, the checkbox and notifications are no longer shown.

CleanShot 2021-06-17 at 12 12 53@2x

With a bit more scope, we could offer the ability to "convert" a saved search notification into a code monitor on the spot. This would take whatever's in the saved search query and would take the user to the "Create new code monitor" view with the saved search notification pre-filled. Then, we could "uncheck/remove" the notification from the saved search while preserving the search as a "useful bookmark."

CleanShot 2021-06-17 at 12 14 56@2x

In this approach, I would rather take the user to the create code monitor view than keep them in place (and create the code monitor behind the scenes), because there's a risk that not all saved search queries might be valid by default, and if something goes wrong it won't be surfaced well. As well, when more actions are available (like webhooks), taking the user to the create view will help surface more options.

Once the user has hit "Convert to code monitor," we'd "uncheck" it and subsequently only show the description and query fields.

Note that the button label has changed and there's now a "Cancel" button as well.

Flow diagram and basic wireframes on Excalidraw: [https://plus.excalidraw.com/l/4Dr1S6qmmY7/9coop4HogrC](https://plus.excalidraw.com/l/4Dr1S6qmmY7/9coop4HogrC)

### Next steps

We need to agree on which approach (or another approach!) we'd like to take. Then, we can finalize our plan.

### PR DESCRIPTION
**Pull request by limitedmage**
Closes #22154

- If notifications for a saved search are **not** enabled: Removes notifications checkbox and show message under form linking users to code monitoring.
- If notifications for a saved search are enabled: Show message under checkbox where users can create a code monitor based on the saved search (name and trigger will be auto-populated).
- Changelog has been updated to note that notifications for saved searches have been deprecated in favor of code monitoring.

image

image

### CODE REVIEW

File: client/web/src/enterprise/code-monitoring/components/CodeMonitorForm.tsx [hunk 1]
```diff
@@ -36,6 +36,8 @@ export interface CodeMonitorFormProps {
     showDeleteButton?: boolean
     /* Optional trigger query to pre-populate the trigger form */
     triggerQuery?: string
+    /* Optional descritption to pre-populate the name */
```
Comment (tjkandala) [inline client/web/src/enterprise/code-monitoring/components/CodeMonitorForm.tsx:39]:
```suggestion
    /* Optional description to pre-populate the name */
```

**Review by tjkandala — APPROVED** [review]

File: client/web/src/enterprise/code-monitoring/components/CodeMonitorForm.tsx (hunk 1 above)
Comment (limitedmage) [reply client/web/src/enterprise/code-monitoring/components/CodeMonitorForm.tsx]: Can I blame GitHub Copilot for this? :P

**Comment by limitedmage** [comment]
@quinnkeast  Can you please double-check UI and copy for this?

**Review by quinnkeast — APPROVED** [review]
Looks rad. This is a great iterative step. Thank you!

**Pull request merged**